### ISSUE TITLE
When activity theme is not a material theme, Beagle view crashes.

### ISSUE THREAD

**Issue by kingori** [pandulapeter/beagle#117]
When trying to show Beagle view on activities which uses non-material theme, app crashes ( Drawer ) or dialog not shown ( Modal).

I think it is caused by using `MaterialButton` class in xml or code.

Is it possible to replace `MaterialButton` to `Button`?

crash log is like below: 
```
java.lang.IllegalArgumentException: The style on this component requires your app theme to be Theme.MaterialComponents (or a descendant).
 at com.google.android.material.internal.ThemeEnforcement.checkTheme(ThemeEnforcement.java:247)
 at com.google.android.material.internal.ThemeEnforcement.checkMaterialTheme(ThemeEnforcement.java:216)
 at com.google.android.material.internal.ThemeEnforcement.checkCompatibleTheme(ThemeEnforcement.java:144)
 at com.google.android.material.internal.ThemeEnforcement.obtainStyledAttributes(ThemeEnforcement.java:76)
 at com.google.android.material.button.MaterialButton.<init>(MaterialButton.java:230)
```

I've assigned non-material theme in `init` function like below:
```
 Beagle.initialize(application, Appearance( mypkg.R.style.My_non_material_theme))
```

**Comment by pandulapeter** [comment]
Hi!
Thanks for bringing this up.
Indeed the current version of the library expects a Material theme.
But you're right, there's no reason for using MaterialButton (if your Activity has a Material theme, the layout inflater will replace all Views with there Material counterparts - something I didn't know back then).
I'll make sure to fix this in the next update, hopefully in the coming days. I'll keep you posted!

**Comment by kingori** [comment]
@pandulapeter thanks for the reply.
If you expect material-theme in this library, what about to wrap context with material-themed context? 

This is code snippet I am using in my project
```kotlin
fun Context.toMaterialContext(): Context {
    // this code is dangerous - using private material theme attribute
    val typedArray = this.theme.obtainStyledAttributes(intArrayOf(com.google.android.material.R.attr.isMaterialTheme))
    val isMaterialTheme = typedArray.getBoolean(0, false)
    typedArray.recycle()

    return if (isMaterialTheme) {
        this
    } else {
        wrapMaterialTheme( R.style.my_default_material_theeme )
        )
    }
}

fun Context.wrapMaterialTheme(@StyleRes newMaterialThemeResId: Int): Context {
    return ContextThemeWrapper(
        this,
        newMaterialThemeResId
    ).apply {
        val materialContext = this
        val delegate = AppCompatDelegate.create(
            materialContext,
            requireNotNull(materialContext.extractActivity()),
            null
        )

        val inflater = LayoutInflater.from(materialContext)
        LayoutInflaterCompat.setFactory2(
            inflater,
            delegate as LayoutInflater.Factory2
        )
    }
}
```

If you apply this code, you can ensure library is using material themed context. But if user is applyed non-material theme, their theme is overridden. But I don't think this cause big(?) problem.

**Comment by jakubturcovsky** [comment]
@pandulapeter Any ETA on this fix?

**Comment by pandulapeter** [comment]
Thanks for the reminder, this issue has been here for a while. This weekend I'll have some time to focus on this, and the next update will include the fix.

**Comment by pandulapeter** [comment]
Fixed in version [2.9.7](https://github.com/pandulapeter/beagle/releases/tag/2.9.7)
If the app doesn't provide a valid Material theme, Beagle now comes with a default one.
If you don't like the default theme, don't forget that you can use something like this to override it:
```
Beagle.initialize(
    application = this,
    appearance = Appearance(
        themeResourceId = R.style.YourMaterialTheme
    ),
    ...
)
```
Thanks @kingori and @jakubturcovsky !
